samples: unicast audio: Wrap start of the send timer

mkasenberg opened the pull request:
Enable the send timer only when at least one ASE source is enabled,
CONFIG_BT_ASCS_ASE_SRC_COUNT > 0 in case of unicast_audio_server,
CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SRC_COUNT > 0 for unicast_audio_client.

Thalley commented:
> A nice way to explore and debug the controller.

Can you elaborate a bit more on this? I'm not sure what advantage these overlays provide based on this short commit message :)

mkasenberg commented:
I updated the PR a little to keep only the really needed overlays. What is left is needed to establish a connection with extended advertising and increase the buffers for LE Audio.

With nrf52_bsim you can reproduce a sync issue I met on real hardware with Zephyr host + controller setup:

For one source configured:
In Central -> Peripheral direction only the first data packet is received by peripheral.
In Peripheral -> Central direction no packet is received.
On both sides no error is seen in logs, the apps keep printing:
```
d_01: @00:00:05.473601  Sending mock data with len 1 on streams[0]
d_01: @00:00:06.473603  Sending mock data with len 2 on streams[0]
d_01: @00:00:07.473633  Sending mock data with len 3 on streams[0]
...
```
and only one print about successfully received packet appears on the Peripheral side:
```
d_00: @00:00:05.486447  Incoming audio on stream 0x80e8960 len 1
```


For two sources configured both apps stop streaming at first packet with a log:
```
d_00: @00:00:06.228650  [00:00:06.228,637] <wrn> bt_hci_core: bt_hci_cmd_send_sync: opcode 0x206e status 0x0c
d_00: @00:00:06.271424  Audio Stream 0x80e89e4 stopped with reason 0x13
```

Thalley commented on samples/bluetooth/unicast_audio_client/src/main.c:
```diff
@@ -1101,8 +1101,10 @@ int main(void)
 
-		/* Start send timer */
-		k_work_schedule(&audio_send_work, K_MSEC(0));
+		if (CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SRC_COUNT) {
+			/* Start send timer */
+			k_work_schedule(&audio_send_work, K_MSEC(0));
+		}
```
This does not seem related to the overlay, and probably should be a different commit

```suggestion
		if (IS_ENABLED(CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SRC_COUNT)) {
			/* Start send timer */
			k_work_schedule(&audio_send_work, K_MSEC(0));
		}
```

mkasenberg replied:
Oops, I set `CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SRC_COUNT=2` and was wondering why sending suddenly stopped working, until I read that: IS_ENABLED(config_macro) returns 1 if config_macro is defined to 1, 0 otherwise (including if config_macro is not defined). Nice prank :)

mkasenberg replied:
Maybe there is some alternative to IS_ENABLED? I can't find it.

Thalley commented on samples/bluetooth/unicast_audio_server/src/main.c:
```diff
@@ -782,7 +782,9 @@ int main(void)
 		printk("Advertising successfully started\n");
 
-		k_work_init_delayable(&audio_send_work, audio_timer_timeout);
+		if (CONFIG_BT_ASCS_ASE_SRC_COUNT) {
+			k_work_init_delayable(&audio_send_work, audio_timer_timeout);
+		}
```
Ditto: Not related to the overlay files. 

```suggestion
		if (IS_ENABLED(CONFIG_BT_ASCS_ASE_SRC_COUNT)) {
			k_work_init_delayable(&audio_send_work, audio_timer_timeout);
		}
```

cvinayak reviewed (dismissed):
We are moving away from having board conf files for Controller, so that users can use explicit overlays when needing to use out-of-tree Controllers. Refer to https://github.com/zephyrproject-rtos/zephyr/pull/58717, and remove board conf files in this PR.

Otherwise changes to sample source code is acceptable for review.

mkasenberg committed:
samples: unicast audio: Wrap start of the send timer

Enable the send timer only when at least one ASE source is enabled,
CONFIG_BT_ASCS_ASE_SRC_COUNT > 0 in case of unicast_audio_server,
CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SRC_COUNT > 0 for unicast_audio_client.

Signed-off-by: Magdalena Kasenberg <magdalena.kasenberg@codecoup.pl>

Thalley commented on samples/bluetooth/unicast_audio_client/src/main.c:
```diff
@@ -1101,8 +1101,10 @@ int main(void)
 
-		/* Start send timer */
-		k_work_schedule(&audio_send_work, K_MSEC(0));
+		if (CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SRC_COUNT) {
```
Good catch on the `IS_ENABLED` - But do consider 
```suggestion
		if (CONFIG_BT_BAP_UNICAST_CLIENT_ASE_SRC_COUNT > 0) {
```

Thalley commented on samples/bluetooth/unicast_audio_server/src/main.c:
```diff
@@ -782,7 +782,10 @@ int main(void)
 		printk("Advertising successfully started\n");
 
-		k_work_init_delayable(&audio_send_work, audio_timer_timeout);
+		if (CONFIG_BT_ASCS_ASE_SRC_COUNT) {
```
```suggestion
		if (CONFIG_BT_ASCS_ASE_SRC_COUNT > 0) {
```

cvinayak commented:
Just noticed that you need to give a better commit tittle and message body...
